crates/nu-plugin/src/plugin/declaration.rs

```diff
@@ -126,10 +127,48 @@ impl Command for PluginDeclaration {
             value => CallInput::Value(value),
         };
 
+        // Fetch the configuration for a plugin
+        //
+        // The `plugin` must match the registered name of a plugin.  For
+        // `register nu_plugin_example` the plugin config lookup uses `"example"`
+        let config = self
+            .filename
+            .file_stem()
+            .and_then(|file| {
+                file.to_string_lossy()
+                    .clone()
+                    .strip_prefix("nu_plugin_")
+                    .map(|name| {
+                        nu_engine::get_config(engine_state, stack)
```

Comment:
Not really sure, can we use `engine_state.get_plugin_config(name)` directly?

Reply:
I cannot get tests to pass without using `nu_engine::get_config()`. It seems that updates to the configuration from `$env.config.plugins = {…}` are stored in the stack and `nu_engine::get_config()` is how those get folded in.